zerver/models/messages.py

```diff
@@ -40,6 +40,29 @@ class AbstractMessage(models.Model):
     # Important for efficient indexes and sharding in multi-realm servers.
     realm = models.ForeignKey(Realm, on_delete=CASCADE)
 
+    class MessageType(models.IntegerChoices):
+        NORMAL = 1
+        RESOLVE_TOPIC_NOTIFICATION = 2
+
+    # Important note: This is not to be confused with what is sometimes
+    # called the message_type in the APIs, CountStats or some variable names.
+    # We intend to rename those to avoid this kind of confusion, but what
+    # they refer to is the type of the recipient of message,
+    # namely private vs stream messages.
+    #
+    # The .type attribute defined here is an entirely different concept
+    # that's meant to distinguish between "normal" messages and some special
+    # kind of messages, such as notification messages that may be sent
+    # by system bots.
+    type = models.PositiveSmallIntegerField(
```

Comment:
This still feels ugly a bit even with this disclaimer, but maybe that's fine if we fix up these other things soonish

Reply:
I can do an editing pass on this, but it seems OK.